### PR TITLE
Config unit tests for CI

### LINKED ISSUE #10: Add CI workflows
Add static checks, unit / integration test workflows

### PR DESCRIPTION
**Pull request by mikealfare**
resolves #10

Setup unit tests to run automatically against `main`

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-adapter/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by mikealfare** [comment]
Requires #27 to work

**Comment by mikealfare** [comment]
Requires #28

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        run: hatch run unit-tests
+        shell: bash
+
+      - name: Publish results
+        uses: ./.github/actions/publish-results
+        if: always()
+        with:
+          source-file: "results.csv"
+          file-name: "unit_results"
+          python-version: ${{ matrix.python-version }}
```
Comment (emmyoop) [inline .github/workflows/unit_tests.yml:49]: Why are you doing this?

File: .changes/unreleased/Under the Hood-20240112-230236.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+kind: Under the Hood
+body: Implement unit testing in CI
```
Comment (emmyoop) [inline .changes/unreleased/Under the Hood-20240112-230236.yaml:2]: Generally we don't have changelogs for CI changes.

**Review by emmyoop — APPROVED** [review]
This looks fine.  Just a question on why you save the results.

We should find out if we need both the matrix for python version in the workflow and the python version tests defined in pyroject.toml.  Not a blocker to getting tests running on your PRs though.

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
```
Comment (emmyoop) [inline .github/workflows/unit_tests.yml:30]:
```suggestion
        uses: actions/checkout@v4
```

File: .changes/unreleased/Under the Hood-20240112-230236.yaml (hunk 2 above)
Comment (mikealfare) [reply .changes/unreleased/Under the Hood-20240112-230236.yaml:2]: Fair, force of habit. We can clean that up afterwards.

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (mikealfare) [reply .github/workflows/unit_tests.yml:49]: I'm mimicking the original workflow. I can remove this if it's not needed.

**Comment by mikealfare** [comment]
>We should find out if we need both the matrix for python version in the workflow and the python version tests defined in pyroject.toml.

Agreed. My concern is that I don't know how to run integration tests in parallel using the hatch env matrix. They would run in parallel on the machine, but not as separate workflows in GHA. We're already running into memory issues for Redshift on Windows, so I want to split across machines. There might be a plugin for hatch that does this, I just haven't looked into it.

**Pull request merged**